### PR TITLE
Move source request to pydevd. Fixes #1333

### LINKED ISSUE #1333: on_source special cases should be ported to pydevd and it should use _forward_request_to_pydevd
Note: `_forward_request_to_pydevd` should `send_error_response` when needed.

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #1333

### CODE REVIEW

File: src/ptvsd/wrapper.py [hunk 1]
```diff
@@ -1423,23 +1423,7 @@ def on_threads(self, request, args):
...
-            _, _, resp_args = yield self.pydevd_request(
-                pydevd_comm.CMD_LOAD_SOURCE,
-                pydevd_request,
-                is_json=True)
-
-            if resp_args.get('success', True):
-                self.send_response(request, **resp_args['body'])
-            else:
-                self.send_error_response(request, resp_args.get('message', 'Error retrieving source'))
+        self._forward_request_to_pydevd(request, args)
```
Comment (fabioz) [inline src/ptvsd/wrapper.py]: This should no longer be `async_handler`.

File: src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py [hunk 2]
```diff
@@ -702,7 +702,9 @@ def on_source_request(self, py_db, request):
         response_args = {'body': body}
 
         if content is None:
-            if server_filename:
+            if source_reference == 0:
```
Comment (fabioz) [inline src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py]: Can you double-check if there are tests for this case? (even if only in the `ptvsd` side)

File: src/ptvsd/wrapper.py (hunk 1 above)
Comment (karthiknadig) [reply src/ptvsd/wrapper.py]: oops

File: src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py (hunk 2 above)
Comment (karthiknadig) [reply src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py]: Updated `test_with_path_mappings` to test this case.

**Pull request merged**